services/src/main/java/io/scalecube/services/RemoteServiceInstance.java

```diff
@@ -37,6 +37,8 @@ public RemoteServiceInstance(ICluster cluster, ServiceReference serviceReference
     this.cluster = cluster;
     this.address = serviceReference.address();
     this.memberId = serviceReference.memberId();
+    this.futures = new ConcurrentHashMap<>();
+    this.cluster.listen().subscribe(message -> handleReply(message));
```

Comment:
1. Still why do you need to make separate subscription for each separate RemoteServiceInstance and keep separate future map for each service instance. Single future map and subscription would be totally enough for the whole system.